tests/integration.rs

```diff
@@ -5992,26 +5992,63 @@ mod common_parallel {
     #[test]
     #[cfg(not(feature = "mshv"))]
     fn test_snapshot_restore_hotplug_virtiomem() {
-        _test_snapshot_restore(true);
+        _test_snapshot_restore(true, false);
     }
 
     #[test]
     fn test_snapshot_restore_basic() {
-        _test_snapshot_restore(false);
+        _test_snapshot_restore(false, false);
     }
 
-    fn _test_snapshot_restore(use_hotplug: bool) {
+    #[test]
+    fn test_snapshot_restore_with_fd() {
```

Comment:
Could you please help me understand why not reuse `_test_snapshot_restore()` ?
The new param added `use_fd`, when set, does not do anything additional than changing the '--net' params.

Reply:
This is to reduce the complexity of tests for better readability. As you see, the `_test_snapshot_restore()` is for testing `_hotplug_virtiomem`, which has nothing to do with the new test `*_with_fd`. While it will have more boilerplate code, having the new test by itself is cleaner and easier to follow for debugging. 

This is what we do for `live_upgrade/migration_*` tests, where tests are reusing logics only if they are related.

Reply:
@likebreath New test now does not use `_test_snapshot_restore()`. But I took out pause, snapshot, event check operations into a common function.